# HG changeset patch
# User Rik <rik@octave.org>
# Date 1428428551 25200
#      Tue Apr 07 10:42:31 2015 -0700
# Node ID 6b78feb92f533879a787e8768fa92c70155ef87f
# Parent  76c72314d9055aa9fa70e2b39fec41ebe65d9a96
Accept, but warn about unimplemented Matlab print options (bug #44760).

* __print_parse_opts__.m: Accept, but Warn about '-opengl', '-painters', and
'-noui' options.

diff --git a/scripts/plot/util/private/__print_parse_opts__.m b/scripts/plot/util/private/__print_parse_opts__.m
--- a/scripts/plot/util/private/__print_parse_opts__.m
+++ b/scripts/plot/util/private/__print_parse_opts__.m
@@ -151,16 +151,20 @@ function arg_st = __print_parse_opts__ (
           arg_st.font = arg(3:end);
         endif
       elseif (length (arg) > 2 && arg(1:2) == "-S")
         arg_st.canvas_size = str2num (arg(3:end));
       elseif (length (arg) > 2 && arg(1:2) == "-r")
         arg_st.ghostscript.resolution = str2double (arg(3:end));
       elseif (length (arg) > 2 && arg(1:2) == "-f")
         arg_st.figure = str2num (arg(3:end));
+      elseif (any (strcmp (arg, {"-painters", "-opengl"})))
+        warning ("print: '%s' accepted for Matlab compatibility, but is ignored", arg);
+      elseif (strcmp (arg, "-noui"))
+        warning ("print: option '-noui' not yet implemented");
       elseif (length (arg) >= 1 && arg(1) == "-")
         error ("print: unknown option '%s'", arg);
       elseif (length (arg) > 0)
         arg_st.name = tilde_expand (arg);
       endif
     elseif (isfigure (arg))
       arg_st.figure = arg;
     else
